sessiontxn/isolation/repeatable_read.go

```diff
@@ -179,24 +182,37 @@ func (p *PessimisticRRTxnContextProvider) AdviseOptimizeWithPlan(val interface{}
 		plan = execute.Plan
 	}
 
-	mayOptimizeForPointGet := false
-	if v, ok := plan.(*plannercore.PhysicalLock); ok {
-		if _, ok := v.Children()[0].(*plannercore.PointGetPlan); ok {
-			mayOptimizeForPointGet = true
-		}
-	} else if v, ok := plan.(*plannercore.Update); ok {
-		if _, ok := v.SelectPlan.(*plannercore.PointGetPlan); ok {
-			mayOptimizeForPointGet = true
+	p.optimizeForNotFetchingLatestTS = optimizeForNotFetchingLatestTS(plan)
+
+	return nil
+}
+
+// optimizeForNotFetchingLatestTS searches for optimization condition recursively
+func optimizeForNotFetchingLatestTS(plan plannercore.Plan) bool {
+	switch v := plan.(type) {
+	case *plannercore.PointGetPlan:
+		return true
+	case *plannercore.BatchPointGetPlan:
+		return true
+	case plannercore.PhysicalPlan:
+		if v.Children() == nil {
+			return false
 		}
-	} else if v, ok := plan.(*plannercore.Delete); ok {
-		if _, ok := v.SelectPlan.(*plannercore.PointGetPlan); ok {
-			mayOptimizeForPointGet = true
+		allChildrenArePointGet := true
+		for _, p := range v.Children() {
+			allChildrenArePointGet = allChildrenArePointGet && optimizeForNotFetchingLatestTS(p)
 		}
+		return allChildrenArePointGet
+	case *plannercore.Update:
+		_, ok := v.SelectPlan.(*plannercore.PointGetPlan)
```

Comment:
How about BatchPointGetPlan ?

Reply:
It did not consider batch point plan before the refactor. But I think it can be added.